### PR TITLE
Remove unused include and library checks

### PR DESCRIPTION
**Pull request by prckent**
## Proposed changes

Per the title, remove several legacy/used include checks and defines

## What type(s) of changes does this code introduce?

- Refactoring (no functional changes, no api changes)
- Build related changes

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

nitrogen, llvm

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- NA. Code added or changed in the PR has been clang-formatted
- NA. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- NA. Documentation has been added (if appropriate)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -220,14 +220,6 @@ INCLUDE(${CMAKE_ROOT}/Modules/CheckTypeSize.cmake)
 
 CHECK_FUNCTION_EXISTS(posix_memalign HAVE_POSIX_MEMALIGN)
 
-CHECK_INCLUDE_FILE(dlfcn.h HAVE_DLFCN_H)
-CHECK_INCLUDE_FILE(inttypes.h HAVE_INTTYPES_H)
```
Comment (ye-luo) [inline CMakeLists.txt:224]: "inttypes.h" still include in some files under src/einspline. Probably all remove.

File: CMakeLists.txt [hunk 2]
```diff
@@ -220,14 +220,6 @@ INCLUDE(${CMAKE_ROOT}/Modules/CheckTypeSize.cmake)
...
-CHECK_INCLUDE_FILE(memory.h HAVE_MEMORY_H)
-CHECK_INCLUDE_FILE(sys/stat.h HAVE_SYS_STAT_H)
-CHECK_INCLUDE_FILE(sys/time.h HAVE_SYS_TIME_H)
```
Comment (ye-luo) [inline CMakeLists.txt:227]: Probably worth keep this one as `sys/time.h` for posix.

File: CMakeLists.txt [hunk 3]
```diff
@@ -220,14 +220,6 @@ INCLUDE(${CMAKE_ROOT}/Modules/CheckTypeSize.cmake)
...
-CHECK_INCLUDE_FILE(sys/types.h HAVE_SYS_TYPES_H)
```
Comment (ye-luo) [inline CMakeLists.txt:228]: Found in `QMCTools/ppconvert/src/ParserClass.cc` probably remove.

File: CMakeLists.txt [hunk 4]
```diff
@@ -220,14 +220,6 @@ INCLUDE(${CMAKE_ROOT}/Modules/CheckTypeSize.cmake)
...
-CHECK_INCLUDE_FILE(unistd.h HAVE_UNISTD_H)
```
Comment (ye-luo) [inline CMakeLists.txt:229]: Can we remove all the unistd.h occurrence?

**Review by ye-luo — CHANGES_REQUESTED** [review]
I think we can purge the leftover.

File: CMakeLists.txt (hunk 1 above)
Comment (prckent) [reply CMakeLists.txt:224]: You are welcome to, I don't have time.

**Comment by prckent** [comment]
This PR is all about removing unused HAVE_* defines and the test for them. Unfortunately I don't have time currently to go through and do further work screening for unused include files.

**Comment by prckent** [comment]
Feel free to purge further.

File: CMakeLists.txt (hunk 4 above)
Comment (prckent) [reply CMakeLists.txt:229]: AFAIK, unistd.h is the preferred route to gethostname. We have a few of those.

File: CMakeLists.txt (hunk 1 above)
Comment (prckent) [reply CMakeLists.txt:224]: I experimented. Used to get intptr_t. Required.

Comment (ye-luo) [reply CMakeLists.txt:224]: Replaced with `<stdint.h>`

File: CMakeLists.txt (hunk 4 above)
Comment (ye-luo) [reply CMakeLists.txt:229]: Seems needed.

**Review by ye-luo — APPROVED** [review]

**Pull request merged**